lib/ansible/modules/dnf5.py

```diff
@@ -0,0 +1,692 @@
+            elif command in {"repos", "repositories"}:
+                query = libdnf5.repo.RepoQuery(base)
+                query.filter_enabled(True)
+                results = [{"repoid": repo.get_id(), "state": "enabled"} for repo in query]
+            else:
+                resolve_spec_settings = libdnf5.base.ResolveSpecSettings()
+                query = libdnf5.rpm.PackageQuery(base)
+                query.resolve_pkg_spec(command, resolve_spec_settings, True)
+                results = [package_to_dict(package) for package in query]
+
+            self.module.exit_json(msg="", results=results, rc=0)
+
+        settings = libdnf5.base.GoalJobSettings()
+        settings.group_with_name = True
+        if self.bugfix or self.security:
+            advisory_query = libdnf5.advisory.AdvisoryQuery(base)
+            types = []
+            if self.bugfix:
+                types.append("bugfix")
+            if self.security:
+                types.append("security")
+            advisory_query.filter_type(types)
+            settings.set_advisory_filter(advisory_query)
+
+        goal = libdnf5.base.Goal(base)
+        results = []
+        if self.names == ["*"] and self.state == "latest":
+            goal.add_rpm_upgrade(settings)
+        elif self.state in {"install", "present"}:
+            for spec in self.names:
+                if not is_spec_installed(base, spec):
+                    if self.update_only:
+                        results.append("Packages providing {} not installed due to update_only specified".format(spec))
+                    else:
+                        goal.add_install(spec, settings)
+        elif self.state == "latest":
+            for spec in self.names:
+                if is_spec_installed(base, spec):
+                    goal.add_upgrade(spec, settings)
+                elif not self.update_only:
+                    goal.add_install(spec, settings)
+                else:
+                    results.append("Packages providing {} not installed due to update_only specified".format(spec))
+        elif self.state in {"absent", "removed"}:
+            for spec in self.names:
+                try:
+                    goal.add_remove(spec, settings)
+                except RuntimeError as e:
+                    self.module.fail_json(msg=str(e), failures=[], rc=1)
+            if self.autoremove:
+                for pkg in get_unneeded_pkgs(base):
+                    goal.add_rpm_remove(pkg, settings)
+
+        goal.set_allow_erasing(self.allowerasing)
+        try:
+            transaction = goal.resolve()
+        except RuntimeError as e:
+            self.module.fail_json(msg=str(e), failures=[], rc=1)
+
+        if transaction.get_problems():
+            failures = []
+            for log in transaction.get_resolve_logs_as_strings():
+                if log.startswith("No match for argument") and self.state in {"install", "present", "latest"}:
```

Comment:
This is something I did to make the CI pass and didn't care for at the time. I'll double check here as well.

Reply:
I see what's the problem here. On C++ API we have a method `get_resolve_logs()` which returns `LogEvent` objects instead of strings. There you can determine related `spec`, `action`, etc. This is not yet available in bindings, I will create an issue for this in DNF5.